### PR TITLE
feat(docs): Add SignServer integration documentation

### PR DESCRIPTION
**Pull request by jiparis**
This PR adds SignServer integration documentation, and some pointers for setting up SignServer service.

### CODE REVIEW

File: docs/docs/guides/deployment/keyfactor/signserver.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+* [Quick Start Guide - Start SignServer Container with Client Certificate Authenticated Access.](https://docs.keyfactor.com/signserver/latest/quick-start-guide-start-signserver-container-with-)
+
+Finally, you must have a Crypto and Signing workers configured in SignServer. You can follow [this Cosign tutorial](https://docs.keyfactor.com/signserver/latest/tutorial-signserver-container-signing-with-cosign), since the steps for worker configuration are similar.
+
+Once you have it configured, you can reach SignServer signer worker at, for example, `https://mysignserver/PlainSigner`.
+
+Note that, at this moment, SignServer authentication is not implemented.
+
+### Signing Chainloop attestations with SignServer
+Using the CLI options above, Chainloop with prepare the signing In-toto payload and send it to SignServer for signing. The returned signature is then
```
Comment (migmartri) [inline docs/docs/guides/deployment/keyfactor/signserver.md:26]: `signing` => `signed`

File: docs/docs/guides/deployment/keyfactor/signserver.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+├───────────────────┼──────────────────────────────────────┤
+│ Attestation ID    │ 32c7eeed-ae4d-4a07-8a85-44a6b05458dc │
+│ Name              │ mywf                                 │
+│ Team              │                                      │
+│ Project           │ myproject                            │
+│ Contract Revision │ 1                                    │
+└───────────────────┴──────────────────────────────────────┘
+
+➜ chainloop att add --value evidence.txt
+WRN API contacted in insecure mode
```
Comment (migmartri) [inline docs/docs/guides/deployment/keyfactor/signserver.md:48]: I'd remove the line `WRN API contacted in insecure mod`

File: docs/docs/guides/deployment/keyfactor/signserver.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+* [Quick Start Guide - Issue Client Authentication Certificate using EJBCA](https://docs.keyfactor.com/ejbca/latest/quick-start-guide-issue-client-authentication-cert)
+* [Quick Start Guide - Start SignServer Container with Client Certificate Authenticated Access.](https://docs.keyfactor.com/signserver/latest/quick-start-guide-start-signserver-container-with-)
+
+Finally, you must have a Crypto and Signing workers configured in SignServer. You can follow [this Cosign tutorial](https://docs.keyfactor.com/signserver/latest/tutorial-signserver-container-signing-with-cosign), since the steps for worker configuration are similar.
+
+Once you have it configured, you can reach SignServer signer worker at, for example, `https://mysignserver/PlainSigner`.
+
+Note that, at this moment, SignServer authentication is not implemented.
+
+### Signing Chainloop attestations with SignServer
```
Comment (migmartri) [inline docs/docs/guides/deployment/keyfactor/signserver.md:25]: We are missing the verification section aren't we?

File: docs/docs/guides/deployment/keyfactor/signserver.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+Using the CLI options above, Chainloop with prepare the signing In-toto payload and send it to SignServer for signing. The returned signature is then 
+included in the final attestation and sent to Chainloop Vault for storage.
+
+![chainloop-signserver-sequence.png](chainloop-signserver-sequence.png)
```
Comment (migmartri) [inline docs/docs/guides/deployment/keyfactor/signserver.md:29]: nice diagram

File: docs/docs/guides/deployment/keyfactor/signserver.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
+---
+title: Use Keyfactor SignServer for attestation signing
+---
+
+### Summary
+Chainloop CLI can sign attestations using a preconfigured SignServer instance, by providing a key reference during the signing process:
```
Comment (migmartri) [inline docs/docs/guides/deployment/keyfactor/signserver.md:6]: I'd also add a paragraph of what Signserver is and a link to their about page https://www.signserver.org/about/

**Review by migmartri — APPROVED** [review]

File: docs/docs/guides/deployment/keyfactor/signserver.md (hunk 3 above)
Comment (jiparis) [reply docs/docs/guides/deployment/keyfactor/signserver.md:25]: I've added it.

**Pull request merged**